compiler/src/dotty/tools/dotc/config/Settings.scala

```diff
@@ -91,38 +88,45 @@ object Settings {
 
     def isDefaultIn(state: SettingsState): Boolean = valueIn(state) == default
 
+    def isMultivalue: Boolean = implicitly[ClassTag[T]] == ListTag
+
     def legalChoices: String =
-      if (choices.isEmpty) ""
-      else choices match {
-        case r: Range => s"${r.head}..${r.last}"
-        case xs: List[?] => xs.mkString(", ")
+      choices match {
+        case xs if xs.isEmpty => ""
+        case r: Range         => s"${r.head}..${r.last}"
+        case xs: List[?]      => xs.toString
       }
 
     def isLegal(arg: Any): Boolean =
-      if (choices.isEmpty)
-        arg match {
-          case _: T => true
-          case _ => false
-        }
-      else choices match {
+      choices match {
+        case xs if xs.isEmpty =>
+          arg match {
+            case _: T => true
+            case _ => false
+          }
         case r: Range =>
           arg match {
             case x: Int => r.head <= x && x <= r.last
             case _ => false
           }
         case xs: List[?] =>
-          xs contains arg
+          xs.contains(arg)
       }
 
     def tryToSet(state: ArgsSummary): ArgsSummary = {
       val ArgsSummary(sstate, arg :: args, errors, warnings) = state
       def update(value: Any, args: List[String]) =
-        if (changed)
-          ArgsSummary(updateIn(sstate, value), args, errors, warnings :+ s"Flag $name set repeatedly")
-        else {
-          changed = true
-          ArgsSummary(updateIn(sstate, value), args, errors, warnings)
-        }
+        val (value1, twicely) =
+          if changed && isMultivalue then
+            val value0  = value.asInstanceOf[List[String]]
+            val current = valueIn(sstate).asInstanceOf[List[String]]
+            val newly   = current ++ value0.filterNot(current.contains)
+            (newly, value0.exists(current.contains))
+          else
+            (value, changed)
+        val dangers = if twicely then warnings :+ s"Flag $name set repeatedly" else warnings
```

Comment:
This warning is not informative for multivalued settings, as it's OK to set them multiple times.

Reply:
Redundant `-language:foo -language:foo` or `-Wconf:cat=deprecation:s -Wconf:cat=deprecation:s` is probably unintended, although possibly harmless. Or, `-Wconf:cat=deprecation:s -Wconf:msg='foo':e -Wconf:cat=deprecation:s` may be ambiguous; or it may be intentional. I'll remove the filter completely instead of guessing.

Reply:
Sorry I wasn't clear. I don't mean to suppress the warning. I mean to improve the warning to say a specific value is set multiple times.